generate/main.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"encoding/csv"
+	"errors"
+	"fmt"
+	u "github.com/ipfs/go-ipfs-util"
+	"io"
+	"math/bits"
+	"net"
+	"os"
+)
+
+const (
+	pkgName        = "asnutil"
+	ipv6OutputFile = "ipv6_asn_map.go"
+	ipv6MapName    = "ipv6CidrToAsnMap"
+)
+
+func main() {
+	// file with the ASN mappings for IPv6 CIDRs.
+	// See ipv6_asn.tsv
+	ipv6File := os.Getenv("ASN_IPV6_FILE")
+
+	if len(ipv6File) == 0 {
+		panic(errors.New("environment vars must be provided"))
+	}
+
+	ipv6CidrToAsnMap := readMappingFile(ipv6File)
+	f, err := os.Create(ipv6OutputFile)
+	if err != nil {
+		panic(err)
+	}
+	defer f.Close()
+	writeMappingToFile(f, ipv6CidrToAsnMap, ipv6MapName)
+}
+
+func writeMappingToFile(f *os.File, m map[string]string, mapName string) {
+	printf := func(s string, args ...interface{}) {
+		_, err := fmt.Fprintf(f, s, args...)
+		if err != nil {
+			panic(err)
+		}
+	}
+	printf("package %s\n\n", pkgName)
```

Comment:
consider `template`, which may make this easier to read:
e.g. https://github.com/ipfs/go-datastore/blob/master/fuzz/cmd/generate/main.go#L60